src/test/java/com/google/api/codegen/testdata/java_test_test_library.baseline

```diff
@@ -0,0 +1,326 @@
+    String formattedName = LibraryServiceApi.formatBookName("[SHELF]", "[BOOK]");
+    Book book = Book.newBuilder().build();
+    api.UpdateBook(formattedName, book);
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    UpdateBookRequest request = (UpdateBookRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+    Assert.assertEquals(request.getBook(), book);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void LibraryServiceTest() {
+    String formattedName = LibraryServiceApi.formatBookName("[SHELF]", "[BOOK]");
+    String formattedOtherShelfName = LibraryServiceApi.formatShelfName("[SHELF]");
+    api.MoveBook(formattedName, formattedOtherShelfName);
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    MoveBookRequest request = (MoveBookRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+    Assert.assertEquals(request.getOtherShelfName(), formattedOtherShelfName);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void LibraryServiceTest() {
+
+    PageAccessor<String> response = api.ListStrings();
+    for (String element : response) {
+      // DO NOTHING
+    }
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    ListStringsRequest request = (ListStringsRequest)requests[0];
+
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void LibraryServiceTest() {
+    String formattedName = LibraryServiceApi.formatBookName("[SHELF]", "[BOOK]");
+    ByteString comment = ByteString.copyFromUtf8("");
+    Comment commentsElement = Comment.newBuilder()
+      .setComment(comment)
+      .build();
+    List<Comment> comments = Arrays.asList(commentsElement);
+    api.AddComments(formattedName, comments);
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    AddCommentsRequest request = (AddCommentsRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+    Assert.assertEquals(request.getComment(), comment);
+    Assert.assertEquals(request.getCommentsElement(), commentsElement);
+    Assert.assertEquals(request.getComments(), comments);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void LibraryServiceTest() {
+    String formattedName = LibraryServiceApi.formatArchivedBookName("[ARCHIVE_PATH]", "[BOOK]");
+    api.GetBookFromArchive(formattedName);
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    GetBookFromArchiveRequest request = (GetBookFromArchiveRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void LibraryServiceTest() {
```

Comment:
I am planing to separate the code into 3 parts: 
1. Test code generation <--- This is the current PR
2. Mock grpc service code generation
3. GAX

I cannot run the test until all 3 pieces are in place. Alternatively, I can work on a branch (maybe test-generator) of this repo instead of pushing into master.